### PR TITLE
Add pin to map for user's current location #21

### PR DESCRIPTION
**Pull request by lbare**

### CODE REVIEW

File: app/src/lib/Map.svelte [hunk 1]
```diff
@@ -50,8 +50,27 @@
         return div
     }
 
+    const getUserLocation = () => {
```
Comment (MarcusDunn) [inline app/src/lib/Map.svelte:53]: rename this to something like `initilizeUserTracking`. `getUserLocation` implies it returns something and doesn't have an effect.

File: app/src/lib/Map.svelte [hunk 2]
```diff
@@ -50,8 +50,27 @@
...
+        setInterval(() => {
```
Comment (MarcusDunn) [inline app/src/lib/Map.svelte:54]: This returns a reference to the interval, you'll want to clean up the infinite loop in the `destroy` callback (line 85).

**Comment by MarcusDunn** [comment]
Looks good, couple of small changes and I think this resolves #21

File: app/src/lib/Map.svelte (hunk 2 above)
Comment (lbare) [reply app/src/lib/Map.svelte:54]: how would I go about this?

Comment (MarcusDunn) [reply app/src/lib/Map.svelte:54]: simplest way would be to have a userLocation component level variable that you assign inside `initilizeUserTracking` then call `clearInterval` on it in destroy

File: app/src/lib/Map.svelte [hunk 3]
```diff
@@ -50,8 +50,32 @@
...
+                timeout: 5000,
+                maximumAge: 0
+            });
+        }, 2000);
+        
+        return {
+            destroy: () =>
+                clearInterval(userLocation)
+        }
+    }
```
Comment (lbare) [inline app/src/lib/Map.svelte:73]: @MarcusDunn does this make sense for clearing the interval?

Comment (MarcusDunn) [reply app/src/lib/Map.svelte:73]: almost. you have to call it in the same function that `map.remove()` is in.

Comment (MarcusDunn) [reply app/src/lib/Map.svelte:73]: change `initializeUserTracking();` to `const stopTrackingUser = initializeUserTracking();`

and change ```return {
            destroy: () => {
                map.remove()
            },
        };```

to ```
return {
            destroy: () => {
                map.remove()
                stopTrackingUser()
            },
        };```

Comment (lbare) [reply app/src/lib/Map.svelte:73]: So do I need to remove the `destroy `from `initializeUserTracking`?

Comment (MarcusDunn) [reply app/src/lib/Map.svelte:73]: if you *only* have it return a function that calls `clearInterval` I think that would be simplest. you could also do (keeping the return as it is now) to `stopTrackingUser.destroy()`

**Pull request merged**